src/sql-parser/src/parser.rs

```diff
@@ -2229,6 +2257,17 @@ impl Parser {
         Ok(data_type)
     }
 
+    /// Parse a SQL data type using `parse_data_type`. If unable to parse an existing
+    /// SQL data type, parse as a custom, user-defined SQL data type.
+    fn parse_data_type_or_custom(&mut self) -> Result<DataType, ParserError> {
+        Ok(if let Ok(typ) = self.parse_data_type() {
+            typ
+        } else {
+            self.prev_token();
+            DataType::Custom(self.parse_identifier()?.to_string())
+        })
+    }
```

Comment:
I think instead you should just change `parse_data_type` to support custom types! Because we want to be able to parse `'{key => val }'::foomap`~